pygmt/src/grdtrack.py

```diff
@@ -283,26 +287,65 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
     ...     points=points, grid=grid, newcolname="bathymetry"
     ... )
     """
+    # pylint: disable=too-many-branches
     if hasattr(points, "columns") and newcolname is None:
         raise GMTInvalidInput("Please pass in a str to 'newcolname'")
 
+    if points is None and kwargs.get("E") is None:
+        raise GMTInvalidInput("Must give 'points' or set 'profile'.")
+
+    if points is not None and kwargs.get("E") is not None:
+        raise GMTInvalidInput("Can't set both 'points' and 'profile'.")
+
+    # Backward compatibility with old parameter order "points, grid".
+    # deprecated_version="0.7.0", remove_version="v0.9.0"
+    is_a_grid = True
+    if not isinstance(grid, (xr.DataArray, xr.Dataset, str)):
+        is_a_grid = False
+    elif isinstance(grid, str):
+        try:
+            xr.open_dataarray(which(grid, download="a"))
+            is_a_grid = True
```

Comment:
Does `pygmt.which` work with local NetCDF files? Also, not sure if I like this if-condition that requires opening a file using `xr.open_dataarray` to check if it is a grid. What about using `grdinfo` instead?

Reply:
Hmm, so `pygmt.grdinfo` produces a blank string for non-grid files. What about a check like:

```python
if pygmt.grdinfo(grid) != "":
    is_a_grid = True
else:
    is_a_grid = False
```
As an aside, we colud also think about making `grdinfo` produce proper Python errors as part of the `grdinfo` refactor (#593).

Reply:
> ```python
> if pygmt.grdinfo(grid) != "":
>     is_a_grid = True
> else:
>     is_a_grid = False
> ```

It may work, but users will see the errors like:
```
grdinfo (gmtapi_import_grid): Not a supported grid format [pygmt/tests/data/points.txt]
[Session pygmt-session (2)]: Error returned from GMT API: GMT_GRID_READ_ERROR (18)
[Session pygmt-session (2)]: Error returned from GMT API: GMT_GRID_READ_ERROR (18)
```

Reply:
Hmm yeah, I tried `pygmt.grdinfo(grid="pygmt/tests/data/points.txt", verbose="q")` which removed the first error line, but still getting two `[Session pygmt-session (2)]: Error returned from GMT API: GMT_GRID_READ_ERROR (18)` errors printed.

Reply:
FYI, I've changed the line to 
```
xr.open_dataarray(which(grid, download="a")).close()
```
to recycle any related resources.